conn.go

```diff
@@ -205,23 +206,29 @@ func (s *Session) connect(ctx context.Context, host *HostInfo, errorHandler Conn
 	return s.dial(ctx, host, s.connCfg, errorHandler)
 }
 
-func (s *Session) connectWithDialer(ctx context.Context, host *HostInfo, errorHandler ConnErrorHandler, dialer Dialer) (*Conn, error) {
-	return s.dialWithDialer(ctx, host, s.connCfg, errorHandler, dialer)
+// connectShard establishes a connection to a shard.
+// if nrShards is zero, shard-aware port is disabled.
```

Comment:
Nitpick: rephrase this line as in the comment for `dialShard`, i.e. it's not that the shard-aware port itself is disabled, but rather dialing to that port.

Reply:
Updated. Thanks!